pallets/randomness/src/lib.rs

```diff
@@ -308,9 +285,8 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(_now: BlockNumberFor<T>) -> Weight {
-			// Do not verify default VrfOutput set in first block
-			if frame_system::Pallet::<T>::block_number().is_zero() {
-				// block number is whitelisted
+			// No VrfOutput set in first block (genesis or runtime upgrade)
+			if LocalVrfOutput::<T>::get().is_none() {
 				return Zero::zero();
 			}
```

Comment:
I thought this would work, but it would just never set the VrfOutput because it's not set in the first block.

The existing solution in the codebase uses a bool which is set in the first block to only avoid verifying for the first block but verifying for all blocks thereafter. Adding the genesis item worked, but it does not work for running chains adding the pallet so I'm just going to abandon this small optimization in favor of better compatibility for adding this pallet to a running chain.